src/modules.js

```diff
@@ -0,0 +1,450 @@
+        let list = JSON.parse(localStorage.getItem('allTasks')); 
+        if ( list == null) {
+            dataModel.allTasks.customProjects.push(project);
+            localStorage.setItem('allTasks', JSON.stringify(dataModel.allTasks))
+        } else {
+            dataModel.allTasks.customProjects.push(project);
+            //localStorage.removeItem('allTasks');
+            localStorage.setItem('allTasks', JSON.stringify(dataModel.allTasks));
+        };
+    };
+
+    const createAddToProject = (img) => {
+        img.onclick = (e) => {
+            let list = JSON.parse(localStorage.getItem('allTasks'));
+            list = list.customProjects
+            if (list.length == 0) {
+                let div = document.createElement('div');
+                div.classList.add('projectAddPopup');
+                div.innerText= `Create a Project first
+                
+                `;
+                img.parentElement.appendChild(div);
+                let buttonC = document.createElement('button');
+                buttonC.classList.add('.cancelTaskToProjectBtn');
+                buttonC.innerText="Cancel";
+                div.appendChild(buttonC);
+                cancelTaskToProjectHandler(buttonC);
+            } else {
+                let div = document.createElement('div');
+                div.classList.add('projectAddPopup');
+                div.innerText=`Select a project to add this task to:
+
+                `;
+                img.parentElement.appendChild(div);
+                let sel = document.createElement("select")
+                for (let i=0; i< list.length; i++) { 
+                    let opt = document.createElement("option");
+                    opt.value = list[i].name; 
+                    opt.text = list[i].name; 
+                    sel.add(opt)
+                }
+                div.appendChild(sel);
+                let buttonS = document.createElement('button');
+                buttonS.classList.add('.addTaskToProjectBtn');
+                buttonS.innerText="Submit";
+                div.appendChild(buttonS);
+                saveTaskToProject(img, buttonS, sel);
+                let button = document.createElement('button');
+                button.classList.add('.cancelTaskToProjectBtn');
+                button.innerText="Cancel";
+                div.appendChild(button);
+                cancelTaskToProjectHandler(button);
+                return sel
+            }
+        }
+    };
+
+    const pageLoad = () => {
+        let taskStorage = JSON.parse(localStorage.getItem('allTasks'));
+        if (taskStorage == null) {
+            return
+        } else {
+            getTasks(taskStorage);
+            getProjects(taskStorage);
+            localStorage.removeItem('allTasks');
+            localStorage.setItem('allTasks', JSON.stringify(dataModel.allTasks));
+        };
+    };
+
+    const getTasks = (taskStorage) => {
+        taskStorage = taskStorage.tasks
```

Comment:
Haha I see what you were going for here. Reducing the number of variables by simple reusing an existing one. This will likely work no issue here, however it's usually bad practice to do this sort of thing. The reasons are complicated, but in the simplest terms overriding an existing variable adds confusions that can easily be avoided. It also loosely falls into the "side-effect" category that I've mentioned previously.